### PR TITLE
#38 confirmation of created schedule

### LINKED ISSUE #38: US-38: As a monitoring user, I would like to get confirmation screen of a new time-location schedule I have created for a supervised user

### PR DESCRIPTION
**Pull request by KennanHoa**
#38 

**Description**
- Confirmation screen when creating a schedule with all the selected information for the schedule.
- User can cancel to modify the data or confirm to create the new schedule.

**How to check**
- Create a new schedule and see if the confirmation screen pops up.

### CODE REVIEW

**Review by refatK — CHANGES_REQUESTED** [review]
image

A test broke due to the change to the form, so the test will have to be fixed. Shouldn't be too complicated.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
Feature works well.

The only problem is that a test for setting schedule is failing, and should be fixed.

**Comment by refatK** [comment]
Code and functionality is good, so just the test needs to be fixed and the branch updated, and it should be good to merge.

File: app/src/main/java/com/example/bigowlapp/model/Schedule.java [hunk 1]
```diff
@@ -111,4 +111,8 @@ public void setLocation(GeoPoint location) {
     public void setUserScheduleResponseMap(Map<String, UserScheduleResponse> userScheduleResponseMap) {
         this.userScheduleResponseMap = userScheduleResponseMap;
     }
+
+    public String toString(){
+        return ("\n\nTitle: "+ getTitle() + "\n\nMembers: " + getMemberList() + "\n\nStart Time: " + getStartTime() + "\n\nEnd Time: " + getEndTime() + "\n\nLocation: " + getLocation());
+    }
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/model/Schedule.java]: Is this needed?

File: app/src/main/java/com/example/bigowlapp/fragments/ScheduleFormFragment.java [hunk 2]
```diff
@@ -407,4 +407,49 @@ public Button getSelectUserButton() {
...
+                Toast.makeText(getContext(), "Schedule added successfully", Toast.LENGTH_LONG).show();
+                getActivity().finish();
+            }
+        });
+
+        scheduleConfirmation.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+            }
+        });
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/fragments/ScheduleFormFragment.java]: Can convert these to lambda calls instead.

File: app/src/main/java/com/example/bigowlapp/model/Schedule.java (hunk 1 above)
Comment (KennanHoa) [reply app/src/main/java/com/example/bigowlapp/model/Schedule.java]: removed

**Review by KhangZhengNg — APPROVED** [review]
Good job adding the confirmation step and the testing file

File: app/src/androidTest/java/com/example/bigowlapp/fragments/ScheduleFormFragmentTest.java [hunk 3]
```diff
@@ -241,6 +243,23 @@ public void completeAScheduleForm() {
...
+        String startDate = dateFormatter(startNewTime);
+        String startTime = timeFormatter(startNewTime);
+        String endDate = dateFormatter(newEndTime);
+        String endTime = timeFormatter(newEndTime);
+        String selectedlocation = (fakeSelectedLocation.placeName() == null) ? location.address() : location.placeName();
+
+        onView(withText("Please confirm the new Schedule info: \n\nTitle:\n" + "Schedule_Title" + "\n\nGroup name:\n"+ "group_name" + "\n\nSelected members:\n"+ selectedMembers
+                + "\nStart date:\n" + startDate + "\n"+ startTime + "\n\nEnd date\n" + endDate + "\n" + endTime + "\n\nSelected Location:\n" + selectedlocation))
+                .check(matches(isDisplayed()));
+        onView(withText("CONFIRM")).perform(click());
```
Comment (refatK) [inline app/src/androidTest/java/com/example/bigowlapp/fragments/ScheduleFormFragmentTest.java:261]: Nice job integrating your additions to the test, both fixing it and increasing coverage a bit

File: app/src/main/java/com/example/bigowlapp/fragments/ScheduleFormFragment.java [hunk 4]
```diff
@@ -447,9 +447,9 @@ public String confirmationString (Group group, List<User> userList, Schedule sch
         String endDate = dateFormatter(endDateTime);
         String endTime = timeFormatter(endDateTime);
 
-        String selectedlocation = (location.placeName() == null) ? location.address() : location.placeName();
+        String selectedLocation = (location.placeName() == null) ? location.address() : location.placeName();
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/fragments/ScheduleFormFragment.java:444]: Nice, good style fix

**Review by refatK — APPROVED** [review]
Feature works, and you addressed my changes, so good jib and approving!

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Pull request merged**